### PR TITLE
Consolidated all documentation into the docs folder

### PR DESCRIPTION
**Pull request by quaat**
All documentation (handwritten and generated) is consolidated into the docs folder
the sphinx project is run from the generated build/doc/_build folder (look here to find the artifacts).
The final output is written to build/doc/html

### CODE REVIEW

File: doc/_templates/respirator/src/dlite.rst.j2 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{% extends "doxygenfile.rst.j2" %}
+{% block description %}
+DDDDDDLITE
```
Comment (jesper-friis) [inline doc/_templates/respirator/src/dlite.rst.j2:3]: Should there be so many D's?

File: doc/_templates/respirator/src/utils.rst.j2 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{% extends "toc.rst.j2" %}
+{% block description %}
+lorem ipsum
```
Comment (jesper-friis) [inline doc/_templates/respirator/src/utils.rst.j2:3]: looks generic. Intended?

File: doc/user_documentation/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/README.md
+    COMMAND ${CMAKE_COMMAND}
+      -Dinfile=${dlite_SOURCE_DIR}/README.md
+      -Doutfile=${CMAKE_CURRENT_BINARY_DIR}/README.md
+      -P ${CMAKE_CURRENT_SOURCE_DIR}/fix_readme.cmake
+    MAIN_DEPENDENCY ${dlite_SOURCE_DIR}/README.md
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/fix_readme.cmake
+    )
+
+  set(dependencies
```
Comment (jesper-friis) [inline doc/user_documentation/CMakeLists.txt:17]: Is it documented somewhere that we have to remember to add new README or header files to this list?

**Comment by jesper-friis** [comment]
I think that tips_and_tricks.md should go to developers documentation rather than users documentation.

File: doc/_templates/respirator/src/dlite.rst.j2 (hunk 1 above)
Comment (quaat) [reply doc/_templates/respirator/src/dlite.rst.j2:3]: It was a test to check if custom comments worked. it did. I will remove this :)

File: doc/_templates/respirator/src/utils.rst.j2 (hunk 2 above)
Comment (quaat) [reply doc/_templates/respirator/src/utils.rst.j2:3]: This has been replaced with a proper description

File: doc/user_documentation/CMakeLists.txt (hunk 3 above)
Comment (quaat) [reply doc/user_documentation/CMakeLists.txt:17]: This has just been copied in from the old doc-folder. I think a better to configure doxygen and autodoc to recognize FIXME-statements or similar, to automatically generate this from the source code.

**Comment by quaat** [comment]
> I think that tips_and_tricks.md should go to developers documentation rather than users documentation.
Well, the final structure is a different issue. This PR has the focus of consolidating the doc and pydoc folders into one unified documentation system. Structuring and styling comes later

**Review by jesper-friis — APPROVED** [review]
I am happy with this. 

Would you make an issue with the suggestions about the structure that we skip in this PR?

**Comment by quaat** [comment]
Consolidated all documentation into the docs folder. All API documentation is generated in the build folder (build/doc/_build).

**Pull request merged**